docs/multi-pages.md

```diff
@@ -0,0 +1,113 @@
+# Multi-page scenarios
+
+Playwright can automate scenarios that span multiple browser contexts or multiple
+tabs in a browser window.
+
+<!-- GEN:toc-top-level -->
+- [Multiple contexts](#multiple-contexts)
+- [Multiple pages](#multiple-pages)
+- [Handling new pages](#handling-new-pages)
+- [Handling popups](#handling-popups)
+<!-- GEN:stop -->
+
+## Multiple contexts
+
+[Browser contexts](core-concepts.md#browser-contexts) are isolated environments
+on a single browser instance. Playwright can create multiple browser contexts
+within a single scenario. This is useful when you want to test for
+multi-user functionality, like chat.
+
+```js
+const { chromium } = require('playwright');
+
+// Create a Chromium browser instance
+const browser = await chromium.launch();
+
+// Create two isolated browser contexts
+const userContext = await browser.newContext();
+const adminContext = await browser.newContext();
+
+// Load user and admin cookies
+await userContext.addCookies(userCookies);
+await adminContext.addCookies(adminCookies);
+```
+
+#### API reference
+
+- [class `BrowserContext`](./api.md#class-browsercontext)
+- [`browser.newContext([options])`](./api.md#browsernewcontextoptions)
+- [`browserContext.addCookies(cookies)`](api.md#browsercontextaddcookiescookies)
+
+## Multiple pages
+
+Each browser context can host multiple pages (tabs).
+
+* Each page behaves like a focused, active page. Bringing the page to front
+  is not required.
+* Pages inside a context respect context-level emulation, like viewport sizes,
+  custom network routes or browser locale.
+
+```js
+// Create two pages
+const pageOne = await context.newPage();
+const pageTwo = await context.newPage();
+
+// Get pages of a brower context
+const allPages = await context.pages();
+```
+
+#### API reference
+
+- [class `Page`](./api.md#class-page)
+- [`browserContext.newPage()`](./api.md#browsercontextnewpage)
+- [`browserContext.pages()`](./api.md#browsercontextpages)
+
+## Handling new pages
+
+The `page` event on browser contexts can be used to get new pages that are
+created in the context. This can be used to handle new pages opened by
+`target="_blank"` links.
+
+```js
+// Get page after a specific action (e.g. clicking a link)
+const [newPage] = await Promise.all([
+  context.waitForEvent('page'),
+  page.evaluate(() => window.open('https://google.com', '_blank'))
+])
+console.log(await newPage.title());
```

Comment:
You'll need at least domcontentloaded before accessing title, but preferably load for everything else.

```js
await newPage.waitForLoadState('domcontentloaded');
```

Reply:
Thanks. Updated to use `newPage.waitForLoadState()` since that covers more scenarios for the audience of a guide like this.